From ea46f6edae059d2f6a83bb8901d9142349a1008a Mon Sep 17 00:00:00 2001
From: Volker Krause <vkrause@kde.org>
Date: Mon, 4 Sep 2023 17:43:13 +0200
Subject: [PATCH] Manually manage permissions in the Android manifest

The automatic permission handling gives us e.g. audio recording here,
due to linking against Qt MultiMedia, which we don't need/use but which
looks suspicious.
---
 src/app/android/AndroidManifest.xml | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/app/android/AndroidManifest.xml b/src/app/android/AndroidManifest.xml
index 4fb5254e..20948339 100644
--- a/src/app/android/AndroidManifest.xml
+++ b/src/app/android/AndroidManifest.xml
@@ -13,8 +13,13 @@
 
     <supports-screens android:largeScreens="true" android:normalScreens="true" android:anyDensity="true" android:smallScreens="true"/>
 
+    <!-- manual permission management, the Qt automatism would add too much here -->
     <uses-permission android:name="android.permission.READ_CALENDAR"/>
-    <!-- %%INSERT_PERMISSIONS -->
+    <uses-permission android:name="android.permission.INTERNET"/>
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/> <!-- from Qt, do we really need that? -->
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/> <!-- for onboard API -->
+    <uses-permission android:name="android.permission.CAMERA"/> <!-- barcode scanner -->
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/> <!-- onboard API, navigation -->
     <!-- %%INSERT_FEATURES -->
 
     <application android:name="org.qtproject.qt5.android.bindings.QtApplication" android:label="@string/application_name_full" android:icon="@mipmap/ic_launcher">
-- 
2.41.0

